proposals/improved-interpolated-strings.md

```diff
@@ -364,6 +364,8 @@ of a larger expression `e`, any components of `e` that occurred before `i` will
 3. If `Fc` ended with a `bool` out argument, a check on that `bool` value is generated. If true, the methods in `Fa` will be called. Otherwise, they will not be called.
 4. For every `Fax` in `Fa`, `Fax` is called on `ib` with either the current literal component or _interpolation_ expression, as appropriate. If `Fax` returns a `bool`, the result is
 logically anded with all preceeding `Fax` calls.
+    1. If `Fax` is a call to `AppendLiteral`, the literal component is unescaped by replacing any _open_brace_escape_sequence_ with a single `{`, and any _close_brace_escape_sequence_
+    with a single `}`.
```

Comment:
How does this impact the first argument passed to the interpolation handler constructor? Essentially consider the string `$"{hello} {{". Do we pass the length 2 or 3 to the handler constructor? 

My instinct is that we pass 2 because the spirit of the length argument is to represent the statically known length of the string. Given that the handler only sees the single brace we should only pass length information for it.

Reply:
We pass 2, but I do probably need to modify that section of the spec as well to account for this.